hypothesis.py

```diff
@@ -77,6 +78,59 @@ def sample_ttest(longitude: float,
 
     return HypothesisTest(longitude, latitude, p_value, p_value_target)
 
+def get_predictions(sample_data: pd.DataFrame,
+                    isotope_column_names: list[str],
+                    means_isoscapes: list[raster.AmazonGeoTiff],
+                    variances_isoscapes: list[raster.AmazonGeoTiff],
+                    sample_size_per_location: int):
+  '''
+  Calculates the p values of a hypothesis test for the elements specified by
+  isotope_column_names using values from means_isoscapes and variances_isoscapes.
+
+  sample_data: pd.DataFrame with lat, long, isotope_value and fraudulent columns
+  means_isoscape: Isoscape that maps geographic coordinates to a mean isotope value.
+  variances_isoscape: Isoscape that maps geographic coordinates to the variance of
+                      isotope valuesat that location.
+  sample_size_per_location: Number of samples per geographic location used to calculate
+                            mean and variance in isoscapes.
+  '''
+  sample_data = sample_data.groupby([
+      _TREE_CODE_COLUMN_NAME,
+      _LONGITUDE_COLUMN_NAME,
+      _LATITUDE_COLUMN_NAME,
+      _FRAUDULENT_COLUMN_NAME])[isotope_column_names]
+  predictions = pd.DataFrame({"Code": [],
+                              "long": [],
+                              "lat": [],
+                              "fraud": [],
+                              "predicted_fraud": []})
+
+  for group_key, isotope_values in sample_data:
+    if isotope_values.shape[0] <= 1:
+      continue
+
+    p_values = []
+    for i, isotope_column_name in enumerate(isotope_column_names):
+      hypothesis_test = sample_ttest(longitude=group_key[1],
+                                     latitude=group_key[2],
+                                     isotope_values=isotope_values[isotope_column_name],
+                                     means_isoscape=means_isoscapes[i],
+                                     variances_isoscape=variances_isoscapes[i],
+                                     sample_size_per_location=sample_size_per_location,
+                                     p_value_target=None)
+      p_values.append(hypothesis_test.p_value)
+    combined_p_value = np.array(p_values).prod()
+    if np.isnan(combined_p_value):
+      continue
+
+    row = {"Code": [group_key[0]],
+           "long": [group_key[1]],
+           "lat": [group_key[2]],
+           "fraud": [group_key[3]],
+           "predicted_fraud": [combined_p_value]}
```

Comment:
Are you aware that a high-value actually means a low chance of fraud in a t-test for equality? This refers to comments I made in the tests and the P-R curve PR.

Reply:
"A low p-value in our t-test indicates that two distributions (the ground truth and sample being tested) are dissimilar, which should cause a positive (fraud) result."
https://screenshot.googleplex.com/8gphW7cydwLeBEB

if p-value is under the threshold, we have a positive result.

Fraud prediction == true and p_value < p_valuethreshold => true_positive